packages/graph/hash_graph/lib/interval-ops/src/interval.rs

```diff
@@ -193,84 +221,1191 @@ pub trait Interval<T>: Sized {
+        // Range B:        [-----] | [-----)
+        // intersection:   [---)   |   [---)
+        // union:        [-------] | [-------]
+        // merge:        [-------] | [-------]
+        // difference:   [-)       |       [-]
+        test(
+            included_excluded(0, 10),
+            included_included(5, 15),
+            [included_excluded(5, 10)],
+            [included_included(0, 15)],
+            [included_included(0, 15)],
+            [included_excluded(0, 5)],
+        );
+        test(
+            included_included(5, 15),
+            included_excluded(0, 10),
+            [included_excluded(5, 10)],
+            [included_included(0, 15)],
+            [included_included(0, 15)],
+            [included_included(10, 15)],
+        );
+
+        // Range A:      [-----]   |   [-----)
+        // Range B:        [-----) | [-----]
+        // intersection:   [---]   |   [---]
+        // union:        [-------) | [-------)
+        // merge:        [-------) | [-------)
+        // difference:   [-)       |       (-)
+        test(
+            included_included(0, 10),
+            included_excluded(5, 15),
+            [included_included(5, 10)],
+            [included_excluded(0, 15)],
+            [included_excluded(0, 15)],
+            [included_excluded(0, 5)],
+        );
+        test(
+            included_excluded(5, 15),
+            included_included(0, 10),
+            [included_included(5, 10)],
+            [included_excluded(0, 15)],
+            [included_excluded(0, 15)],
+            [excluded_excluded(10, 15)],
+        );
+
+        // Range A:      [-----)   |   [-----)
+        // Range B:        [-----) | [-----)
+        // intersection:   [---)   |   [---)
+        // union:        [-------) | [-------)
+        // merge:        [-------) | [-------)
+        // difference:   [-)       |       [-)
+        test(
+            included_excluded(0, 10),
+            included_excluded(5, 15),
+            [included_excluded(5, 10)],
+            [included_excluded(0, 15)],
+            [included_excluded(0, 15)],
+            [included_excluded(0, 5)],
+        );
+        test(
+            included_excluded(5, 15),
+            included_excluded(0, 10),
+            [included_excluded(5, 10)],
+            [included_excluded(0, 15)],
+            [included_excluded(0, 15)],
+            [included_excluded(10, 15)],
+        );
+
+        // Range A:      (-----]   |   [-----]
+        // Range B:        [-----] | (-----]
+        // intersection:   [---]   |   [---]
+        // union:        (-------] | (-------]
+        // merge:        (-------] | (-------]
+        // difference:   (-)       |       (-]
+        test(
+            excluded_included(0, 10),
+            included_included(5, 15),
+            [included_included(5, 10)],
+            [excluded_included(0, 15)],
+            [excluded_included(0, 15)],
+            [excluded_excluded(0, 5)],
```

Comment:
I guess the same result here as above. I suppose it does make sense as we'd exclude the lower bound of range B, which would make the upper bound of the resulting interval open ended while the beginning of range A is open ended

Reply:
Yes, it's the same as above, but for the lower bounds